### ISSUE TITLE
[REVIEW]: shar: A R package to analyze species-habitat associations using point pattern analysis

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3811]
**Submitting author:** @mhesselbarth (<a href="http://orcid.org/0000-0003-1125-9918">Maximilian H.K. Hesselbarth</a>)
**Repository:** <a href="https://github.com/r-spatialecology/shar" target ="_blank">https://github.com/r-spatialecology/shar</a>
**Version:** v1.3
**Editor:** @Bisaloo
**Reviewer:** @lionel68, @tretherington
**Archive:** <a href="https://doi.org/10.5281/zenodo.5761583" target="_blank">10.5281/zenodo.5761583</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b/status.svg)](https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b"><img src="https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b/status.svg)](https://joss.theoj.org/papers/1b786c028a5425858cb0e5428bd9173b)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/r-spatialecology/shar" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@lionel68 & @tretherington, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Bisaloo know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @lionel68
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/r-spatialecology/shar">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/r-spatialecology/shar">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mhesselbarth) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @tretherington
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/r-spatialecology/shar">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/r-spatialecology/shar">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mhesselbarth) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by Bisaloo** [comment]
:wave: :wave: :wave: @mhesselbarth @lionel68 @tretherington this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#3811` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 4-6 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@bisaloo) if you have any questions/concerns.

**Comment by lionel68** [comment]
Ok just made a quick first go through the manuscript and still have to check the functionalities of the package itself so right now I will focus on the text:

1. The species-habitat association derived from point pattern as used in `shar` are based on null models, there are however other ways to estimate these associations by directly modelling the intensity of the point pattern to habitat / covariates, I am thinking of [inlabru::lgcp](https://inlabru-org.github.io/inlabru/articles/web/2d_lgcp_covars.html) or [spatstat::kppm](https://www.rdocumentation.org/packages/spatstat/versions/1.64-1/topics/kppm). These other, more direct, approaches to estimate species-habitat association should be mentioned in the text
2.  The text is in some part a bit confusing and could be improved:
- Line 8: "Studying species-habitat associations is one tool to reveal ... ", I find it weird to refer to studies as tools, maybe replace tool with "approach" or similar?
- Line 15-19: I was a bit confused in this part about the ecological level, I usually envision point pattern analysis as being done within a species with **individuals** clustering in certain habitats and environmental conditions. Analysis of species-habitat associations is usually done within a species (as is the case in this package), I found these sentences therefore confusing concerning which levels where under interest here.
- Line 26-28: The definition of point pattern is rather terse, I would expand and add concrete examples to make it clearer to the readers which datasets would qualify as a point pattern and which not.

3. It would be interesting to have some ideas on which raster / point pattern size would give results in a reasonable amount of time with the package. Like how long does it takes to run 10'000 simulations of a raster with 1Mio pixels? And are there options / functionalities built in the package to run these computations on large dataset efficiently (parallel computing, GPU ...)

Voila that's it for now, I'll look at the package R functionalities in the next days.

**Comment by mhesselbarth** [comment]
@lionel68 @tretherington
Thanks for the helpful and good comments. I will start working on them and push changes to the `joss` branch!

**Comment by mhesselbarth** [comment]
Okay, I think from my side I tried to address all comments by @tretherington and @lionel68.

**Comment by tretherington** [comment]
Sorry, I do have other thoughts, but have been swamped by a few tasks I've had to prioritise, but I'm trying to reserve some time tomorrow afternoon to provide more feedback to keep things moving along.

**Comment by mhesselbarth** [comment]
Issues that still need to be addressed:

- [ ] r-spatialecology/shar#79
- [ ] r-spatialecology/shar#87

**Comment by mhesselbarth** [comment]
@Bisaloo  
I am planning to use v1.3 for the next release with all the helpful comments and suggestions by the reviewers. The first @whedon comment in this thread says v1.2.2. Not sure if thats important.

**Comment by tretherington** [comment]
I have just tried to run over all the points again, and for me I think the only outstanding one might be: "State of the field: Do the authors describe how this software compares to other commonly-used packages?"

At the moment the paper doesn't refer to any other packages that might be commonly used to do species-habitat analyses.  I acknowledge that the package is quite possibly unique in that it does habitat analyses that account for the spatial patterns in the occurrence and habitat data, as opposed to more classical methods that just bin the data and assume randomness.  But I do wonder if it would be helpful to make reference to some commonly used R packages that apply these more simplistic (perhaps incorrect!?) techniques as I think it could help a potential user to create a cognitive connection between how they currently do things, and how they could do things better by using `shar`.

For example, if I were to read that a function in a package I commonly use might be inappropriate then I think that is more likely to grab my attention as that provides a massive indicator of relevance.  So perhaps at the end of the first paragraph of the statement of need section in the paper where you describe the sub-optimal approach you could give some examples of packages that include that sub-optimal functionality?

The problem I have here is that the species distribution/ecological niche modelling I tend to do is not really comparable to this approach - as I use continuous variables to produce a continuous model rather than a table of results from categorical data.  So I'm unable to readily identify R packages that do similar habitat analyses.  From vague memory I think the `adehabitatHS` package might be a good example as it includes a `compana` function for compositional analysis of habitat use that I think would qualify as an example of an approach that assumes statistical independence of the observations.  But after that I am stuck, but perhaps you will have a better idea given you have more experience in this area.

Apologies that this isn't a very specific comment 😬 but hopefully that makes some sense?

**Comment by Bisaloo** [comment]
> I am planning to use v1.3 for the next release with all the helpful comments and suggestions by the reviewers. The first @whedon comment in this thread says v1.2.2. Not sure if thats important.

That's not a problem. Feel free to increment the version number. We ask you to do it right before acceptance anyways. I'll tell whedon what version should be marked as accepted later.

**Comment by mhesselbarth** [comment]
@tretherington 
https://github.com/r-spatialecology/shar/commit/3850fb113935544b61fa033e637cba45e0ebb365

I changed the last 1,2 paragraphs of the statement of need section quite a bit trying to compare `shar` more to other existing packages and methods. 

However, I don't think that including `adehabitatHS` makes too much sense, since it is not really based on a spatially explicit point pattern approach, but rather on matrices of habitat use.

**Comment by tretherington** [comment]
Thanks for this.  I understand that there may not be anything comparable in terms of methods to `shar`, but I guess I was thinking that if everyone is using a package that uses the equivalent suboptimal methods, then pointing out that limitation would be helpful.  But you are the subject expert here, so I'm happy to go with what you have done. 😄

**Comment by tretherington** [comment]
@Bisaloo I've now ticked off everything in my reviewer checklist 🥳 but as a first time reviewer for JOSS if I've missed a step please let me know.

**Comment by Bisaloo** [comment]
> @Bisaloo I've now ticked off everything in my reviewer checklist partying_face but as a first time reviewer for JOSS if I've missed a step please let me know.

@tretherington, you're all good. Thank you so much for your very helpful comments! :100:

**Comment by Bisaloo** [comment]
@lionel68, could you please go over you checklist in [this comment](https://github.com/openjournals/joss-reviews/issues/3811#issue-1022291832)?
After reviewing each item, either check it or open an issue in https://github.com/r-spatialecology/shar/issues if you think it should be improved. Thanks :pray:

**Comment by mhesselbarth** [comment]
@lionel68 
Also, feel free to re-open any issues I already closed! Here is a list with all your already closed issue: https://github.com/openjournals/joss-reviews/issues/3811#issuecomment-948515516

**Comment by lionel68** [comment]
Sorry for the long wait, will look at this this afternoon, @mhesselbarth is there a new article proof, would be easier for me than going through the md file.

**Comment by Bisaloo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3811 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by Bisaloo** [comment]
@whedon generate pdf from branch joss

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: :point_left:

**Comment by lionel68** [comment]
Thanks @whedon 

Some further / last minor comments:

1. line 24: I would rephrase, "Spatial autocorrelation of the environmental conditions could also ..."
2. line 27: this sentence is a bit cryptic regarding the method developed in the package, maybe best to directly name it? Like "Previous research have shown that violation of independence assumptions led to higher number of false-positive habitat associations when using method XX compared to method YY"
3. line 104: didn't function `reconstruct_pattern` changed in previous commit [beec06e](https://github.com/r-spatialecology/shar/commit/beec06e422a22088450cceacfe4d62871e438d4a) ?

Otherwise looks good!

**Comment by mhesselbarth** [comment]
- [ ]  line 24: I would rephrase, "Spatial autocorrelation of the environmental conditions could also ..."
- [ ]  line 27: this sentence is a bit cryptic regarding the method developed in the package, maybe best to directly name it? Like "Previous research have shown that violation of independence assumptions led to higher number of false-positive habitat associations when using method XX compared to method YY"
- [ ]  line 104: didn't function reconstruct_pattern changed in previous commit beec06e ?

**Comment by mhesselbarth** [comment]
Fixed in r-spatialecology/shar@75c571b188d03ec82fcb48b6bc5f3ab23fe28063

**Comment by Bisaloo** [comment]
@lionel68, you didn't check "State of the field: Do the authors describe how this software compares to other commonly-used packages?". 

Is this on purpose or did you miss it?

**Comment by lionel68** [comment]
My bad, sorry for not checking this thread than often, but seems good to go now! Congrats @mhesselbarth for your careful and patient improvements.

**Comment by mhesselbarth** [comment]
@lionel68 @tretherington 
Thanks for all the valuable input!

**Comment by Bisaloo** [comment]
Great! Thanks @lionel68 and @tretherington for your valuable comments :pray: .

@mhesselbarth, is everything ready on your side? If so, I'll quickly go through everything one last time today and recommend this paper for acceptance. The editor in chief on rotation will take of the rest after.

**Comment by Bisaloo** [comment]
Yes please :+1:, it will probably be easier to review everything

**Comment by Bisaloo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: :point_left:

**Comment by Bisaloo** [comment]
- I fixed a couple of typos in the manuscript in https://github.com/r-spatialecology/shar/pull/91
- It's a good idea to update your GitHub Actions from upstream from time to time. For example, you could have benefit from the new `clean = TRUE` argument in `pkgdown::deploy_to_branch()`, which would probably have helped a little with your pkgdown problems yesterday.
- If you choose to list the dependencies in the Installation section of your README, you should probably add Rcpp which is a non-base dependency as well

**Comment by mhesselbarth** [comment]
https://github.com/r-spatialecology/shar/commit/e49029242d54503381105fc2594eb95ee511f60c

- I accepted the PR
- I added the `clean = TRUE` argument, but I think thats also the default (?)
- I removed `Rcpp` from the `DESCRIPTION` since `shar` no longer depends on it.

**Comment by mhesselbarth** [comment]
All GitHub actions currently fail because https://packagemanager.rstudio.com/ is down (https://github.com/r-lib/actions/issues/447).

**Comment by Bisaloo** [comment]
What about the reference written as plain text? Yamada et al. 2006

**Comment by mhesselbarth** [comment]
https://github.com/r-spatialecology/shar/commit/5ba1f84ead10cd15856826700acc665ee4980799
Ups...thanks for pointing out! That completly slipped my mind.

**Comment by Bisaloo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: :point_left:

**Comment by Bisaloo** [comment]
Can you please have one last look at the paper for any last remaining typos?

If everything looks good, please tag a new version on GitHub and I can accept your submission :slightly_smiling_face:

**Comment by mhesselbarth** [comment]
Fixed some last typos in https://github.com/r-spatialecology/shar/commit/bc93f1a21bb87a8d2472830d9a5fc910ea05a629

In the footer my name is abbreviated by "Hesselbarth, M.H.", however, this should be "Hesselbarth, M.H.K."! Is it possible to fix that somehow?

**Comment by Bisaloo** [comment]
Could you try to add a space between `H.` and `K.` [in the yaml header](https://github.com/r-spatialecology/shar/blob/bc93f1a21bb87a8d2472830d9a5fc910ea05a629/paper.md?plain=1#L12) please to make it clear to LaTeX that they are two different words?

I'll dig more and ask around if it doesn't work.

**Comment by Bisaloo** [comment]
If it doesn't work, for reference, initials are defined here:

https://github.com/openjournals/whedon/blob/92346a0773a4003bf0ef361b661dc2787f492936/lib/whedon/author.rb#L46-L48

**Comment by mhesselbarth** [comment]
@whedon generate pdf

**Comment by mhesselbarth** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03811/joss.03811/10.21105.joss.03811.pdf) :page_facing_up: :point_left:

**Comment by mhesselbarth** [comment]
Did not work, but also not really important I think.

I created a tag called JOSS (https://github.com/r-spatialecology/shar/releases/tag/JOSS)

**Comment by Bisaloo** [comment]
@openjournals/dev, it looks like whedon is not able to handle very well authors with multiple middle names. Is there anything that can be done?

**Comment by arfon** [comment]
> @openjournals/dev, it looks like whedon is not able to handle very well authors with multiple middle names. Is there anything that can be done?

This will have to be fixed post publishing sorry. Please tag me again here when the paper is published and I'll fix it up.

**Comment by Bisaloo** [comment]
Okay, so we can proceed @mhesselbarth. At this point could you:

- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by Bisaloo** [comment]
@whedon set v1.3 as version

**Comment by whedon** [comment]
OK. v1.3 is the version.

**Comment by mhesselbarth** [comment]
https://doi.org/10.5281/zenodo.5761583

**Comment by Bisaloo** [comment]
Run @whedon set 10.5281/zenodo.5761583 as archive.

**Comment by Bisaloo** [comment]
@whedon set 10.5281/zenodo.5761583 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5761583" target="_blank">10.5281/zenodo.5761583</a> is the archive.

**Comment by Bisaloo** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/2041-210X.13168 is OK
- 10.1111/j.1365-2745.2007.01229.x is OK
- 10.1038/s41598-017-15789-5 is OK
- 10.1093/femsec/fiu034 is OK
- 10.1007/s00477-011-0541-1 is OK
- 10.1002/ecs2.1847 is OK
- 10.1111/jvs.12090 is OK
- 10.1111/j.1365-2745.2008.01377.x is OK
- 10.1111/ecog.00911 is OK
- 10.1093/jpe/rtw057 is OK
- 10.1111/j.1365-2745.2001.00615.x is OK
- 10.1073/pnas.0604666104 is OK
- 10.1126/science.220.4598.671 is OK
- 10.1111/j.1600-0587.2000.tb00280.x is OK
- 10.1006/jtbi.2000.2158 is OK
- 10.1111/ecog.01579 is OK
- 10.1111/j.1654-1103.2006.tb02479.x is OK
- 10.1371/journal.pone.0156326 is OK

MISSING DOIs

- None

INVALID DOIs

- 10.32614/RJ-2021-048 is INVALID
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2796

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2796, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

looks good!

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1467911765979967488) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2797
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03811](https://doi.org/10.21105/joss.03811)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by Bisaloo** [comment]
Congratulations @mhesselbarth! And thank you so much again for your time and your valuable insight @tretherington and @lionel68!

**Comment by mhesselbarth** [comment]
Thanks everyone! 🥳

**Comment by danielskatz** [comment]
Congratulations to @mhesselbarth (Maximilian H.K. Hesselbarth)!!

And thanks to @Bisaloo for editing, and @lionel68 and @tretherington for reviewing!
We couldn't do this without you

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03811/status.svg)](https://doi.org/10.21105/joss.03811)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03811">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03811/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03811/status.svg
   :target: https://doi.org/10.21105/joss.03811
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03811/status.svg)](https://doi.org/10.21105/joss.03811)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by mhesselbarth** [comment]
@arfon

Reminder that in the footer my name is abbreviated by "Hesselbarth, M. H. [...]", however, it should be "Hesselbarth, M. H. K. [...]".

**Comment by arfon** [comment]
> Reminder that in the footer my name is abbreviated by "Hesselbarth, M. H. [...]", however, it should be "Hesselbarth, M. H. K. [...]".
 

That should be updated now. Please note the PDF on the JOSS site can take up to 24 hours to update due to caching.
